### PR TITLE
Xcode 13 support + iOS 15 fixes

### PR DESCRIPTION
**Pull request by leandroalonso**
Adds support to Xcode 13 (including CI) and fixes issues for iOS 15 + Xcode 13 combo.

Included issues:

* #17225
* #17227
* #17233
* #17379

### To test

1. Run the app
2. Make sure the app has the same look 'n feel
3. Make sure CI is 🟢 

## Regression Notes
1. Potential unintended areas of impact
The whole project

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by jkmassel — APPROVED** [review]
LGTM from the code side of things

**Review by mokagio — COMMENTED** [review]
@jkmassel @leandroalonso LGTM, too. I opened a PR against this branch to ensure we use the same Ruby version across the board: #17418. I think we should wait till that one is merged in here before merging into `develop`.

File: .buildkite/commands/build-for-testing.sh
Comment (AliSoftware) [inline .buildkite/commands/build-for-testing.sh]: I fear that since those comments are hidden in the separate `.sh` files and thus not noticable directly from the `.yml` pipelines, we might miss them next time we update the image.

What do you think about doing something like this:

1. Add some hashtag to these comments, like:
```suggestion
# FIXIT-13.1: Temporary fix until we're on the Xcode 13.1 VM
```
2. Add a reference to those comment in the 2 `.yml` files:
```yml
    IMAGE_ID: xcode-13 # Remember to address all 'FIXIT-13.1' comments when we'll update to new image
```

**Review by AliSoftware — APPROVED** [review]
I've now merged #17418, so this should now be ready to merge.

I left a comment suggestion to ensure we won't forget to remove the "temporary fixes until Xcode 13.1" next time we update the `IMAGE_ID` (because I feel we'd otherwise miss it); take it or leave it 🙃

**Comment by mokagio** [comment]
We sent out an internal communication (ref. pbArwn-38I) announcing the official Xcode 13 support and will merge this PR on November 9, just to give folks a couple of days to read the announcement and download Xcode 13 if they haven't done so yet.

**Pull request merged**
